core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -51,7 +53,119 @@
 logger.addFilter(dup_filter)
 
 
-class AbstractModel:
+class Taggable(ABC):
```

Comment:
I understand the reasoning behind `Tunable`, but why do we need `Taggable` to be a separate interface instead of defining `_class_tags`, `_more_tags`, etc as method of `ModelBase`?

Reply:
In this case, for reuse. If we can reasonably imagine non-`ModelBase` objects as being taggable one day, it makes sense to me to separate them now.